service/src/main/java/org/mskcc/cmo/metadb/service/impl/SampleServiceImpl.java

```diff
@@ -41,6 +41,8 @@ public MetaDbSample saveSampleMetadata(MetaDbSample
                     updatedMetaDbSample.getPatient().getCmoPatientId().getPatientId());
             if (patient != null) {
                 updatedMetaDbSample.setPatient(patient);
+            } else {
+                patientRepository.save(updatedMetaDbSample.getPatient());
```

Comment:
Still need to save this to a new object

Reply:
need to return foundSample in the else block below (in the case that the foundSample exists)

Reply:
after implementing and replacing the use of the `patientRepository` here with the `patientService` have the service return the updated metadb patient obj and set THAT as the `updatedMetadbSample.setPatient(updatedPatient)`